[Doc] Update the docs about the mount() method of Twig components (#2843)
symfony/ux: src/TwigComponent/doc/index.rst

Mounting Data
-------------

Most of the time, you will create public properties and pass values to them
as "props" when rendering the component. However, there are several hooks

> available when you need to perform more complex logic.

Reviewer: 👍🏻

The mount() Method
~~~~~~~~~~~~~~~~~~

The ``mount()`` method gives you more control over how your "props" are handled.

> It is called only once: immediately after your component is instantiated.

Reviewer: 👍🏻

The ``mount()`` method **cannot access the values of the public properties**

> passed when rendering the component.

Reviewer: I see what you mean, but this is _wrong_ if I want to be picky:
1. you can access the public properties values, if they are not typed or have a default values
2. speaking about only "public properties values" can be misleading, I feel like we should speak about the "props" too in priority

        public string $message;
        public string $type = 'success';

        public function mount(): void
        {
            {# ❌ this won't work: at this point, the $type property still has its
                   default value. Passed properties are not yet available. #}
            if ('error' === $this->type) {
                // ...
            }
        }

        // ...
    }

> Instead, define the values you need in the ``mount()`` method as arguments::

Reviewer: The parameters you define in `mount()` does not only refer to your Component's public properties, but can be any props. For example, if I use `<twig:Alert toto="tata">`, then I can retrieve its value with `public function mount(string $toto)`:

```suggestion
Instead, the props values are available in the ``mount()`` method as arguments::
```

    #[AsTwigComponent]
    class Alert
    {
        public string $message;
        public string $type = 'success';

        public function mount(bool $isError = false): void
        {
            {# ✅ this works as expected #}
            if ($isError) {

>                 // ...

Reviewer: Like it was done before, we should keep an example (or document) that we can modify a property here
```suggestion
                $this->type = 'danger';
```

            }
        }

        // ...
    }

Now, pass the ``mount()`` argument like any other prop:

.. code-block:: html+twig

>     <twig:Alert isError="{{ true }}" message="..."/>

Reviewer: Can we move it just under `Instead, define the values you need in the ``mount()`` method as arguments::`? I feel like the actual sections order can be misleading
